src/core/reactor.cc

```diff
@@ -2075,6 +2075,32 @@ static auto next_waitpid_timeout(std::chrono::milliseconds this_timeout) {
 
 #endif
 
+future<int> reactor::do_waitpid(pid_t pid) {
+    return do_with(int{}, std::chrono::milliseconds(0), [pid, this](int& wstatus,
```

Comment:
Please use coroutines in new code.

Reply:
Ah, you're just moving old code around. ok.

Reply:
https://github.com/scylladb/seastar/pull/2532